docs/source/best-practices/checkpointing.rst

```diff
@@ -74,8 +86,53 @@ To submit this array job to the scheduler, use the following command::
 
    sbatch submit_mdrun_array.bash
 
-Python TensorFlow checkpointing example
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Checkpointing using DMTCP
+--------------------------
+
+`DMTCP <https://dmtcp.sourceforge.io/>`_ (Distributed MultiThreaded checkpointing) is a tool that lets you checkpoint without the need to change your code. It Works with most Linux applications such as Python, Matlab, R, GUI, MPI, etc.
+The program runs in the background of your program without significant performance loss and saves the process states into checkpoint files. DMTCP is available on the cluster ::
+
+ module avail dmtcp
+ module show dmtcp
+ module load dmtcp/2.6.0
+
+As DMTCP runs in the background, it requires some changes to your Shell script. For examples of how to checkpoint with DMTCP visit `here <https://github.com/northeastern-rc/training-checkpointing/tree/main/Exercise_3>`_.
+The example demonstrates how to use DMTCP with a simple C++ program (scripts modified from `RSE-Cambridge <https://github.com/RSE-Cambridge/dmtcp-tests>`_).
+
+Tips and Tricks
+---------------------
+
+What data to save?
+ * Non-temporary application data
+ * Any application data that has changed since the last checkpoint
+ * Delete no longer proper checkpoints - keep only the most recent checkpoint file.
+
+How frequently should we checkpoint?
+ * Too often – will slow down your calculation, maybe I/O heavy and memory-limited.
+ * Too infrequently – leads to large/long rollback times.
+ * Consider how long it takes to a checkpoint and restart your calculation.
+ * In most cases, a rate of every 10-15 minutes is ok.
+
+.. _ML Model-level:
+
+ML Model-level checkpointing
+============================
+
+Checkpointing is a technique used in training machine learning models to save the model's state during the training process periodically. Doing this allows the training process to be resumed from the saved checkpoint if the training is interrupted or terminated so that the progress made so far is not lost. The saved state typically includes the model's parameters, optimizer state, and training information, such as the epoch number and loss value (or the accuracy).
```

Comment:
Consider being more specific with the term being defined, Model - level checkpointing, rather than just checkpointing to be more clear for readers that might be jumping sections. I know it's in the heading, but it's better to be as specific as possible.